App.tsx

```diff
@@ -1,21 +1,102 @@
-import { StatusBar } from 'expo-status-bar';
-import React from 'react';
-import { SafeAreaProvider } from 'react-native-safe-area-context';
+import {
+  DarkTheme,
+  DefaultTheme,
+  NavigationContainer,
+} from "@react-navigation/native";
+import { StatusBar } from "expo-status-bar";
+import React, { useEffect, useState } from "react";
+import { SafeAreaProvider } from "react-native-safe-area-context";
 
-import useCachedResources from './hooks/useCachedResources';
-import useColorScheme from './hooks/useColorScheme';
-import Navigation from './navigation';
+import useCachedResources from "./hooks/useCachedResources";
+import useColorScheme from "./hooks/useColorScheme";
+import MainNavigator from "./navigation/MainNavigator";
+import LinkingConfiguration from "./navigation/LinkingConfiguration";
+import { NavigationScreens } from "./navigation/MainNavigationContext";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { STORAGE_KEYS } from "./utils/AsyncStorageUtils";
+import { AppStateStorage } from "./types";
+import {
+  areDaysEqual,
+  getCurrentDate,
+  retrieveRecordingDay,
+} from "./utils/TimeUtils";
+import { AppState } from "react-native";
 
 export default function App() {
   const isLoadingComplete = useCachedResources();
   const colorScheme = useColorScheme();
+  const [startingState, setStartingState] = useState<AppStateStorage | null>(
+    null
+  );
 
-  if (!isLoadingComplete) {
+  // When getting the state from async storage, the date property is still stored as a string. This fixes that
+  const fixDates = (state: AppStateStorage): AppStateStorage => {
+    return {
+      ...state,
+      date: new Date(state.date),
+    };
+  };
```

Comment:
Since this is a pure function, I think it can be pulled out of the function component (exist at the module level)